crates/pink/sidevm/examples/timer/src/lib.rs

```diff
@@ -8,19 +11,38 @@ use sidevm::{logger::Logger, ocall};
 async fn main() {
     Logger::with_max_level(log::Level::Trace).init();
     ocall::enable_ocall_trace(true).unwrap();
-
+    info!("creating channel...");
+    let (tx, mut rx) = channel(100);
     info!("starting...");
+    for tid in 0..20 {
+        let mut tx = tx.clone();
+        sidevm::spawn(async move {
+            let mut futures: FuturesUnordered<_> = (0..3)
+                .map(|i| async move {
+                    sidevm::time::sleep(std::time::Duration::from_millis(i * 10)).await;
+                })
+                .collect();
 
-    let _ = sidevm::env::spawn(async {
-        for _ in 0..10 {
-            info!("Task 1 sleeping...");
-            sidevm::time::sleep(std::time::Duration::from_secs(1)).await;
-        }
-        info!("Task 1 done");
-    });
-
+            loop {
```

Comment:
Is it possible to join the futures directly?

Reply:
Yes, it can. Because the FuturesUnordered implements the `trait Stream`. And the [`StreamExt`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.collect) has a method `.collect()`.